### ISSUE TITLE
Unreachable hosts causing run_command timeout to be sum of all timeouts

### ISSUE THREAD

**Issue by pkittenis** [ParallelSSH/parallel-ssh#156]
### Bug reports

First reported on mailing list a couple times with no successful reproduction. Have since tracked down the issue. Related to #119 

__Steps to reproduce:__

```
client = ParallelSSHClient(['host1', 'host2', 'host3'])
client.run_command('echo me')
```

In the case where hosts 2 and 3 are unreachable, total time for `run_command` is host2 timeout + host3 timeout.

__Expected behaviour:__ Total timeout is max(host2_timeout, host3_timeout).

__Actual behaviour:__ Total timeout is sum of all individual host timeouts.

__Additional info:__ Latest versions.

The issue is currently `run_command` does a `.get()` on each spawned greenlet sequentially to get output to put in output dictionary. Commands are run in parallel, however `.get` has its own timeout on a per-greenlet basis making total timeout time effectively be the sum of all greenlet timeouts. `run_command` should be calling `joinall` instead.

It should be possible to first do a `joinall` to wait for completion in parallel, followed by `.get` on each greenlet to get output or exception. This will avoid the need for API changes.

It would be good to make it possible to return the successful and unsuccessful greenlets at the same time sans output dictionary, to be made default in 2.x release. This probably as a separate PR.

**Comment by chopraaa** [comment]
Can confirm that this issue exists and is reproducible by adding dummy hosts.

For `hosts = ['host1', 'host2', 'host3', 'dummy_host']`, running a command and usin `join()` takes over 400 seconds with the default timeout.

**Comment by pkittenis** [comment]
Checking if a host is reachable before trying to use it in client can be done outside library. Non reachable hosts can then be excluded from host list. This can be done in the usual methods, `socket.connect` with timeout on socket etc. 

Eg for a parallel version:

```
def _connect(host):
    s = gevent.socket.socket(gevent.socket.AF_INET, gevent.socket.SOCK_STREAM)
    s.timeout = 1
    s.connect((host, 22))
    s.close()
    return host

hosts = [<..>]

conns = [gevent.spawn(_connect, host) for host in hosts]
working_hosts = [s.get() for s in gevent.joinall(conns)]
```

**Comment by chopraaa** [comment]
I initially tried setting the gevent timeout to 1 but I couldn't catch the Timeout exceptions using the library but it's possible to do it here. 

Thanks! This works well.

**Comment by pkittenis** [comment]
Resolved by `1.10`
